modin/core/dataframe/pandas/dataframe/protocol/column.py

```diff
@@ -0,0 +1,590 @@
+
+        Notes
+        -----
+        - Kind specifiers are aligned with DLPack where possible
+          (hence the jump to 20, leave enough room for future extension).
+        - Masks must be specified as boolean with either bit width 1 (for bit masks)
+          or 8 (for byte masks).
+        - Dtype width in bits was preferred over bytes
+        - Endianness isn't too useful, but included now in case in the future
+          we need to support non-native endianness
+        - Went with Apache Arrow format strings over NumPy format strings
+          because they're more complete from a dataframe perspective
+        - Format strings are mostly useful for datetime specification, and for categoricals.
+        - For categoricals, the format string describes the type of the categorical
+          in the data buffer. In case of a separate encoding of the categorical
+          (e.g. an integer to string mapping), this can be derived from ``self.describe_categorical``.
+        - Data types not included: complex, Arrow-style null, binary, decimal,
+          and nested (list, struct, map, union) dtypes.
+        """
+        dtype = self._col.dtypes[0]
+
+        # For now, assume that, if the column dtype is 'O' (i.e., `object`), then we have an array of strings
+        if not isinstance(dtype, pd.CategoricalDtype) and dtype.kind == "O":
+            return (DTypeKind.STRING, 8, "u", "=")
+
+        return self._dtype_from_pandas_dtype(dtype)
+
+    def _dtype_from_pandas_dtype(self, dtype) -> Tuple[DTypeKind, int, str, str]:
+        """
+        Deduce dtype from pandas dtype.
+
+        See `self.dtype` for details.
+
+        Parameters
+        ----------
+        dtype : any
+            A pandas dtype.
+
+        Returns
+        -------
+        tuple
+        """
+        # Note: 'c' (complex) not handled yet (not in array spec v1).
+        #       'b', 'B' (bytes), 'S', 'a', (old-style string) 'V' (void) not handled
+        #       datetime and timedelta both map to datetime (is timedelta handled?)
+        _k = DTypeKind
+        _np_kinds = {
+            "i": _k.INT,
+            "u": _k.UINT,
+            "f": _k.FLOAT,
+            "b": _k.BOOL,
+            "U": _k.STRING,
+            "M": _k.DATETIME,
+            "m": _k.DATETIME,
+        }
+        kind = _np_kinds.get(dtype.kind, None)
+        if kind is None:
+            # Not a NumPy dtype. Check if it's a categorical maybe
+            if isinstance(dtype, pd.CategoricalDtype):
+                # 23 matches CATEGORICAL type in DTypeKind
+                kind = 23
+            else:
+                raise ValueError(
+                    f"Data type {dtype} not supported by exchange protocol"
+                )
+
+        if kind not in (_k.INT, _k.UINT, _k.FLOAT, _k.BOOL, _k.CATEGORICAL, _k.STRING):
+            raise NotImplementedError(f"Data type {dtype} not handled yet")
+
+        bitwidth = dtype.itemsize * 8
+        format_str = dtype.str
```

Comment:
This is not an Apache Arrow C string, [`dtype.str`](https://numpy.org/doc/stable/reference/generated/numpy.dtype.str.html#numpy.dtype.str) returns format string in an array protocol notation when the [dataframe protocol requires](https://data-apis.org/dataframe-protocol/latest/API.html#:~:text=Format%20string%20%3A%20data%20type%20description%20format%20string%20in%20Apache%20Arrow%20C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20Data%20Interface%20format.) it to be in an [Apache Arrow notation](https://arrow.apache.org/docs/format/CDataInterface.html#data-type-description-format-strings).

Reply:
Probably, the function converting numpy type into Arrow notation is the one that has to be located in the `base/protocol/utils.py`

Reply:
fixed